### PR TITLE
geolite-legacy: 20230901 → 20240720

### PR DESCRIPTION
**Pull request by bertptrs**
Update geolite-legacy from version 20230901 to 20240720 for improved accuracy.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/misc/geolite-legacy/default.nix [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
 stdenv.mkDerivation rec {
   pname = "geolite-legacy";
-  version = "20230901";
+  version = "20240720";
 
   # We use Arch Linux package as a snapshot, because upstream database is updated in-place.
   geoip = fetchurl {
     url = "https://archive.archlinux.org/packages/g/geoip-database/geoip-database-${version}-1-any.pkg.tar.zst";
-    sha256 = "sha256-H6tv0OEf04TvbhbWsm5vwq+lBj4GSyOezd258VOT8yQ=";
+    sha256 = "sha256-cokCfGfz1aoJLdR+s+CxDnqkxJ+xFosZ/zwGvAOeZyY=";
```
Comment (itepastra) [inline pkgs/data/misc/geolite-legacy/default.nix:10]: Should use `hash` instead

File: pkgs/data/misc/geolite-legacy/default.nix [hunk 2]
```diff
@@ -2,17 +2,17 @@
 
 stdenv.mkDerivation rec {
   pname = "geolite-legacy";
-  version = "20230901";
+  version = "20240720";
 
   # We use Arch Linux package as a snapshot, because upstream database is updated in-place.
   geoip = fetchurl {
```
Comment (itepastra) [inline pkgs/data/misc/geolite-legacy/default.nix:8]: If you use `fetchzip` instead the tar commands from build should become unnecessary

**Comment by bertptrs** [comment]
Thanks for the quick review! I applied your suggestions.

File: pkgs/data/misc/geolite-legacy/default.nix [hunk 3]
```diff
@@ -1,27 +1,28 @@
-{ lib, stdenv, fetchurl, zstd }:
+{ lib, stdenv, fetchzip, zstd }:
```
Comment (itepastra) [inline pkgs/data/misc/geolite-legacy/default.nix:1]: I notice it wasn't formatted according to nixfmt-rfc-style yet before.

But can you add a second commit where you run nixfmt on the file?

File: pkgs/data/misc/geolite-legacy/default.nix [hunk 4]
```diff
@@ -1,27 +1,28 @@
...
-    tar -xaf ${geoip}
-    tar -xaf ${extra}
-    mkdir -p $out/share
-    mv usr/share/GeoIP $out/share
+    mkdir -p $out/share/GeoIP
+    cp ${geoip}/usr/share/GeoIP/*.dat $out/share/GeoIP
+    cp ${extra}/usr/share/GeoIP/*.dat $out/share/GeoIP
   '';
 
   meta = with lib; {
```
Comment (itepastra) [inline pkgs/data/misc/geolite-legacy/default.nix:28]: This `with lib;` is now considered bad practice iirc, can you remove it and just write `lib.` where it's necessary

**Review by itepastra — APPROVED** [review]

**Comment by itepastra** [comment]
Some nitpicking but shouldn't be blocking

**Comment by bertptrs** [comment]
Also addressed, thanks!

File: pkgs/data/misc/geolite-legacy/default.nix [hunk 5]
```diff
@@ -1,34 +1,40 @@
...
-  meta = with lib; {
+  meta = {
     description = "GeoLite Legacy IP geolocation databases";
     homepage = "https://mailfud.org/geoip-legacy/";
-    license = licenses.cc-by-sa-40;
-    platforms = platforms.all;
-    maintainers = with maintainers; [ fpletz ];
+    license = lib.licenses.cc-by-sa-40;
+    platforms = lib.platforms.all;
+    maintainers = [ lib.maintainers.fpletz ];
```
Comment (itepastra) [inline pkgs/data/misc/geolite-legacy/default.nix:38]: You can do this instead, keeps it more readable when more maintainers are added
```suggestion
    maintainers = with lib.maintainers; [ fpletz ];
```

**Comment by bertptrs** [comment]
Last one?

**Review by itepastra — APPROVED** [review]
LGTM

**Review by fpletz — APPROVED** [review]

**Pull request merged**